### PR TITLE
fix(argo-events): Added missing NATS version to `values.yaml`

### PR DESCRIPTION
**Pull request by rakhbari**
Fixes issue #1724 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by rakhbari** [comment]
I'm a little baffled as to why the helm-docs check is failing. I looked at the `chart-test` output and the `helm-docs.sh` script is exiting with a non-zero exit code. But when I run this script locally, it runs fine (exit code 0). So I'll need one of you helm-docs gurus to let me know what (if anything) I might have missed, perhaps in `README.md.gotmpl`? Or perhaps in `values.yaml` add a `@default` comment for `configs.nats.versions`?

**Comment by egze** [comment]
@rakhbari I checkout your branch and ran `scripts/helm-docs.sh`, was fine.

To make the linter pass, you can add the following to `charts/argo-events/README.md`

```
| configs.nats.versions | list | `[{"metricsExporterImage":"natsio/prometheus-nats-exporter:latest","natsStreamingImage":"nats-streaming:latest","version":"latest"},{"metricsExporterImage":"natsio/prometheus-nats-exporter:0.8.0","natsStreamingImage":"nats-streaming:0.22.1","version":"0.22.1"}]` | Supported versions of NATS event bus |
```

Replace whatever is the line with `| configs.nats.versions | ...` with the above.

File: charts/argo-events/Chart.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ maintainers:
 annotations:
   artifacthub.io/changes: |
     - "[Added]: Ability to specify port for webhook deployment"
+    - "[Fixed]: Missing NATS version in values.yaml"
```
Comment (egze) [inline charts/argo-events/Chart.yaml:19]: Looking at other pull requests and [docs here](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog), it seems you need to always provide just the latest change.

```suggestion
    - "[Fixed]: Missing NATS version in values.yaml"
```

**Comment by egze** [comment]
Only thing missing now is the [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md)

File: charts/argo-events/Chart.yaml (hunk 1 above)
Comment (rakhbari) [reply charts/argo-events/Chart.yaml]: @egze Thanks for the fix to the linter in `argo-events/README.md`.
About the suggestion you made about `argo-events/Chart.yaml`, I accepted your suggestion and used `Sign off and commit` right here in the PR UI, but now the DCO check fails. I thought the `Sign off and commit` would satisfy DCO requirements, but apparently it doesn't. So what do I need to do now? Re-commit locally with `git -s -m "..."` and re-push?

Comment (egze) [reply charts/argo-events/Chart.yaml]: Yes, you can try it.

**Comment by pdrastil** [comment]
Hi @rakhbari, thank you for contribution. The version is intentionally omitted in defaults as the value is used by EventBus resources. If someone bumps this (and it will happen as it happened in the past) expect you will get broken during the upgrade when someone overwrites this. You can see discussion about this in PR https://github.com/argoproj/argo-helm/pull/1465. It's expected that user will provide and pin his versions to avoid this.

**Comment by rakhbari** [comment]
@pdrastil Thanks for the additional context into this issue. It's understandable that you're worried about a broken `EventBus` deployment when the `argoproj/argo-events` dev team bumps their default version. But the point is that this Helm chart, left as is, already results in a broken `EventBus` deployment because the actual version used by the `argo-events` team is not in the list of supported versions. Wouldn't it be better to have a *working* Helm chart for now, and if the `argo-events` dev team bumps their default version, well, the Helm chart simply adds that new default version to its list of supported versions at that point.

The RCA time it took to chase down this issue was a significant sink of my time and I was just trying to avoid anyone else running into the same thing. As I mentioned in the referenced issue #1724, several people have ran into this same issue and even more time & productivity lost, which would be avoided if this default version would simply be added to the Helm chart values.yaml.

But alas, I'll yield to the consensus and decision of the community. Let me know and I can just close this PR unresolved.

**Comment by pdrastil** [comment]
@rakhbari After some thought I'm ok to merge this as it might help other people. Even then I would recommend users to pin their versions.

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.5.5
 kubeVersion: ">=1.22.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 5.16.7
+version: 5.16.9
```
Comment (pdrastil) [inline charts/argo-cd/Chart.yaml:6]: @rakhbari Can you please rebase this? Argo CD changes should not be part of this PR.

Comment (rakhbari) [reply charts/argo-cd/Chart.yaml:6]: @pdrastil: The changes to `argo-cd` and `argo-workflows` charts came in when I was trying to resolved the DCO `sign-off` issue, which instructed me to do `git rebase HEAD~8 --signoff`. But now those commits have been pushed to my fork and therefore part of this PR. I'm pretty sure a `rebase` isn't going to solve it. I think I'm going to have to do `git checkout upstream/master --` on my fork for both `argo-cd/` and `argo-workflows/` folders, then re-commit to remove them from this PR. Do you have a better set of steps to remove those commits from this PR?

Comment (egze) [reply charts/argo-cd/Chart.yaml:6]: Sorry to jump in like that :) I think this is a good plan.

P.S. I'm also waiting for the PR 🚀

Comment (pdrastil) [reply charts/argo-cd/Chart.yaml:6]: What I do in such case is to sync my main with upstream repo and then rebase the PR branch on top.
